modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C

```diff
@@ -77,10 +92,18 @@ ComputeReturnMappingStress::computeQpStress()
   RankTwoTensor strain_increment = _strain_increment[_qp];
   RankTwoTensor stress_new;
   updateQpStress(strain_increment, stress_new);
-  _elastic_strain[_qp] = _rotation_increment[_qp] * (strain_increment + _elastic_strain_old[_qp]) *
-                         _rotation_increment[_qp].transpose();
-  _stress[_qp] = _rotation_increment[_qp] * stress_new * _rotation_increment[_qp].transpose();
 
+  if (_perform_finite_strain_rotations)
+  {
+    _elastic_strain[_qp] = _rotation_increment[_qp] *
```

Comment:
The elastic strain and stress already get rotated forward in `ComputeFiniteStrain.C`.  I guess we would need to rotate the plastic strain forward here.

 I'm guessing the motivation for making this rotation optional is to allow for this to work in a small strain setting.  Is that right?  I wonder if there is anything else we need to change for that to work.  We have really focused on finite deformation for these models.